eth_abi/decoding.py

```diff
@@ -41,40 +41,53 @@ class HeadTailDecoder(BaseDecoder):
 
     def validate(self):
         super().validate()
+
         if self.tail_decoder is None:
             raise ValueError("No `tail_decoder` set")
 
     def decode(self, stream):
         start_pos = decode_uint_256(stream)
         anchor_pos = stream.tell()
+
         stream.seek(start_pos)
         value = self.tail_decoder(stream)
         stream.seek(anchor_pos)
+
         return value
 
 
 class MultiDecoder(BaseDecoder):
     decoders = None
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.decoders = tuple(
+            (
+                HeadTailDecoder(tail_decoder=decoder)
+                if isinstance(decoder, (DynamicArrayDecoder, StringDecoder))
```

Comment:
Wondering if this should instead be a property check like `if decoder.is_dynamic`.  Otherwise 3rd party decoders won't be able to properly integrate right?

Reply:
Yeah I just noticed that actually.  Furthermore, any tuple type which contains at least one dynamic type becomes, itself, a dynamic type.  Since tuples can be nested, I need to check whether or not the coders in a multi coder are dynamic tuple coders.